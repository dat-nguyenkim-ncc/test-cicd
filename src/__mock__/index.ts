import { TagGroupType } from './../types/index.d'
import {
  CompanyBusiness,
  CompanyDetail,
  CompanyFinancials,
  CompanyOverview,
  CompanyPeople,
  MappingSummary,
  TagData,
  TagDataParent,
} from '../types'
import { EnumAttachmentType, EnumCompanyTypeSector } from '../types/enums'

const getRandomId = (): string => {
  return (~~(Math.random() * 999999) + new Date().valueOf()).toString()
}

const companyTypes = Object.values(EnumCompanyTypeSector)
const suffix = ['Restaurant', 'Limited', 'UK', 'US', 'Bank']
export const mockCompanyId = getRandomId()
export const companyDetails = (): CompanyDetail => ({
  companyId: mockCompanyId,
  companyName: `Monzo ${suffix[~~(Math.random() * suffix.length)]}`,
  countryCode: 'GB',
  countryName: 'United Kingdom',
  url: 'https://monzo.com',
  companyTypes: [companyTypes[~~(Math.random() * companyTypes.length)]],
})

export const MockMonzoDetails = companyDetails()

export const mappingSummary = [
  {
    title: 'CATEGORIES',
    items: [
      { label: 'Fintech', parent: [] },
      { label: 'Retail Accounts' },
      { label: 'Accounts & Savings' },
      { label: 'Digital Banking' },
    ],
    type: EnumCompanyTypeSector.FIN,
  },
  {
    title: 'SECTORS',
    items: [{ label: 'No sector mapped', parent: [] }],
    type: EnumCompanyTypeSector.FIN,
  },
] as MappingSummary[]

export const companySources = [
  {
    company: MockMonzoDetails,
    source: {
      label: 'Source: XPTO',
      default: true,
    },
  },
  {
    company: companyDetails(),
    source: {
      label: 'Manual',
    },
  },
]

export const companyOverview = (): CompanyOverview => ({
  id: getRandomId(),
  source: 'CRB',
  sources: ['CRB', 'DR', 'MANUAL'],
  companyName: `Monzo ${suffix[~~(Math.random() * suffix.length)]}`,
  companyLocation: [
    {
      id: '123456',
      address: 'Broadwalk House<br>5 Appold Street<br>London<br>EC2A 2AG',
      postalCode: '94103',
      location: {
        region: 'EMEA',
        city: 'San Francisco',
        country: 'United States',
      },
      isHeadQuarter: true,
      selfDeclared: true,
      expandStatus: 'Following',
      source: 'bcg',
    },
  ],
  description: 'Monzo is a fintech challenger bank.',
  companyType: 'Startup',
  foundedYear: 2012,
  status: 'The company status',
  expandStatus: 'Following',
  expandStatusId: '1',
  closedDate: new Date(2012, 11, 1).toISOString(),
  ftse: 1000,
  logoUrl: 'https://monzo.com/logo.svg',
  contactEmail: 'monzo@monzo.com',
  phoneNumber: '0123456789',
  tags: null,
  otherNames: null,
  lastFundingType: null,
  logo_bucket_url: '',
  hashed_image: '',
  attachments: [
    {
      name: 'File nam',
      description: 'Broadwalk House<br>5 Appold Street<br>London<br>EC2A 2AG',
      type: EnumAttachmentType.FCT_PROFILE,
      url_attachment: '',
      expandStatus: 'Following',
      selfDeclared: true,
      date_created: '2021-03-30 10:43:33',
    },
  ],
  aliases: [
    {
      alias_id: '200354bcg1',
      company_alias: 'name-edit',
      expand_status_id: 1,
      selfDeclared: true,
      source: 'bcg',
    },
    {
      alias_id: '200354bcg2',
      company_alias: 'test1',
      expand_status_id: 1,
      selfDeclared: true,
      source: 'bcg',
    },
  ],
})

export const companyFinancials = (): CompanyFinancials => ({
  valuation: {
    value: 23123123,
    currency: 'USD',
  },
  fundingTotal: {
    value: 2312312312,
    currency: 'USD',
  },
  companyStage: 'Series E+',
  lastFundingDate: {
    value: 1231233,
    currency: 'USD',
  },
  equityFundingTotal: {
    value: 1231233,
    currency: 'USD',
  },
  leadInvestor: 'Mussum Cassilds-Cepacol',
  lastFundingAmount: {
    value: 123123123,
    currency: 'USD',
  },
  fundingRounds: [
    {
      id: '1',
      date: new Date(2012, 11, 1).toISOString(),
      quarter: 'Q2',
      comment: 'This is comment',
      roundTypes: ['Equity Financing, series E+'],
      expandRound1: 'Equity Financing',
      originExpandRound1: 'Equity Financing',
      valuation: {
        value: 123123123,
        currency: 'USD',
      },
      investment: {
        value: 123123123,
        currency: 'USD',
      },
      investors: {
        lead: 'Accel Partners, VC',
        other:
          'Goodwater Capital, VC Orange, Corporate Passion Capital, VC Reference Capital (formerly Genevest), VC Stripe, Tech Thrive Capital, VC Vanderbilt University, Other Y Combinator, Accelerator',
      },
    },
    {
      id: '2',
      date: new Date(2012, 11, 1).toISOString(),
      quarter: 'Q2',
      comment: 'This is comment',
      roundTypes: ['Equity Financing, series E+'],
      expandRound1: 'Equity Financing',
      originExpandRound1: 'Equity Financing',
      valuation: {
        value: 123123123,
        currency: 'USD',
      },
      investment: {
        value: 123123123,
        currency: 'USD',
      },
      investors: {
        lead: 'Accel Partners, VC',
      },
    },
    {
      id: '3',
      date: new Date(2012, 11, 1).toISOString(),
      quarter: 'Q2',
      comment: 'This is comment',
      roundTypes: ['Equity Financing, series E+'],
      expandRound1: 'Equity Financing',
      originExpandRound1: 'Equity Financing',
      valuation: {
        value: 123123123,
        currency: 'USD',
      },
      investment: {
        value: 123123123,
        currency: 'USD',
      },
      investors: {
        lead: 'Accel Partners, VC',
      },
    },
  ],
  isExternalViewDetail: true,
})

export const companyBusiness = (): CompanyBusiness => ({
  productsServices: ['Offers a current account and debit card '],
  businessRevenue: ['Service type: B2C'],
  targetClients: ['Retail customers'],
  partnerships: [
    '**Deliveroo:** Partnered with Deliveroo in Sep-2017 to allow users to split the bill on orders via its Monzo.me money request feature',
  ],
  vision: ['string'],
  keyMetrics: ['string'],
  differentiators: ['string'],
})

export const companyAcquisitions = () => []
export const companyIpos = () => []
export const companyPeople = (): CompanyPeople => ({ companyId: 145803 })

const tags = [
  'Capital Markets',
  'Corporate Banking',
  'Retail Banking',
  'SME Banking',
  'Support',
  'Technology',
  'Wealth Management',
]

const getRandomValueLabel = (index?: number) => {
  const tag = index ? tags[index] : tags[~~(Math.random() * tags.length)]
  return {
    label: tag,
    value: tag.toLowerCase().replace(' ', '_'),
  }
}

const getRandomBoolean = () => {
  return Math.random() * 10 > 5
}

export const TagMock = (
  index?: number,
  level: number = 0,
  parent: TagDataParent[] = []
): TagData => {
  const { value, label } = getRandomValueLabel(index)
  const id = getRandomId()
  const rowId = getRandomId()
  const isPriority = getRandomBoolean()
  return {
    id,
    rowId,
    value,
    label,
    isPriority,
    endpoint: level > 0 ? Math.random() > 0.5 : false,
    parent,
    description: 'Mock',
    children:
      level >= 5
        ? undefined
        : new Array(Math.max(3, ~~(Math.random() * tags.length)))
            .fill(null)
            .map((_a, index) =>
              TagMock(index, level + 1, [...parent, { id, rowId, label, value, isPriority }])
            ),
  }
}

export const ListRandomTags = (): TagData[] =>
  new Array(tags.length).fill(null).map((_a, index) => TagMock(index, 0))

const labels = [
  'Insurance Value Chain 1',
  'Insure Tech Tag',
  'Revenue model',
  'P&C Commercial Customers',
  'Product',
  'RegTech',
  'Regulation',
  'Report',
]

export const TagGroupMock = (): TagGroupType[] =>
  new Array(labels.length * 2).fill(null).map((_a, index) => {
    const id = getRandomId()
    const rowId = getRandomId()
    const label = labels[index % labels.length]
    const value = label.toLowerCase().split(' ').join('_')
    const isPriority = getRandomBoolean()
    return {
      id,
      rowId,
      label,
      value,
      isPriority,
      parent: [],
      children: new Array(Math.max(1, ~~(Math.random() * labels.length * 2)))
        .fill(null)
        .map((_b, index) => {
          const labelChild = labels[index % labels.length]
          return {
            label: labelChild,
            id: getRandomId(),
            rowId: getRandomId(),
            parent: [{ id, label, value }],
            isPriority,
            value: labelChild.toLowerCase().split(' ').join('_'),
          }
        }),
    }
  })
