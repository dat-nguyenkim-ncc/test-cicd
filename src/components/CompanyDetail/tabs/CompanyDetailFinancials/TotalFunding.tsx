import React from 'react'
import { groupBy } from 'lodash'
import { Box, Grid } from 'theme-ui'
import CompanyDetailInline from '../CompanyDetailInline'
import Decimal from 'decimal.js-light'
import { CompanyFinancialsFundingRound, ViewInterface } from '../../../../types'
import { formatMoney } from '../../../../utils'
import strings from '../../../../strings'

type TotalFundingProps = ViewInterface<{
  isExternalViewDetail: boolean
  fundingRounds: CompanyFinancialsFundingRound[]
}>

function TotalFunding({ isExternalViewDetail, fundingRounds }: TotalFundingProps) {
  const {
    companyDetail: { financials: copy },
  } = strings

  if (isExternalViewDetail) return null

  const groupedFundingRounds = groupBy(fundingRounds || [], f => f.expandRound1)
  const totalFunding = fundingRounds
    ?.map(item => item?.investment.value || 0)
    .reduce(
      (pre: number, cur: number) => new Decimal(pre || 0).plus(new Decimal(cur || 0)).toNumber(),
      0
    )

  const getTotalByFundingRounds = (arr: CompanyFinancialsFundingRound[]) => {
    const flatArr = arr?.map(item => item?.investment.value)
    if (flatArr?.every(item => typeof item !== 'number')) return copy.fundingRounds.notAvailable
    const total = flatArr?.reduce(
      (pre: number, cur: number) => new Decimal(pre || 0).plus(new Decimal(cur || 0)).toNumber(),
      0
    )
    if (total) return formatMoney(total, arr[0]?.investment?.currency)
    return copy.fundingRounds.notAvailable
  }

  return (
    <Box mb={3}>
      <CompanyDetailInline
        title={`${copy.fundingRounds.totalFunding}: `}
        key={copy.fundingRounds.totalFunding}
        detail={
          totalFunding ? formatMoney(totalFunding, fundingRounds[0]?.investment?.currency) : copy.fundingRounds.notAvailable
        }
      />
      <Grid columns={'1fr 1fr'}>
        {Object.values(groupedFundingRounds).map((arr, i) => (
          <CompanyDetailInline
            key={arr[0]?.expandRound1 + i}
            title={`TOTAL ${arr[0]?.expandRound1?.toUpperCase()}:`}
            detail={getTotalByFundingRounds(arr)}
          />
        ))}
      </Grid>
    </Box>
  )
}

export default TotalFunding
