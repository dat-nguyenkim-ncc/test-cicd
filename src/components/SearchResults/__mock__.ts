export const data = {
  external: [
    {
      //replace property id by company_id
      //TODO: change all records
      company_id: '7b4545e6-a34d-47c2-853e-ff9eb40c18ef',
      source: 'CRB',
      companyDetails: {
        companyId: '9185a077-c260-ac40-2678-fef9e645591b',
        companyName: 'Monzo',
        url: 'https://monzo.com/',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
    {
      id: 'a67f6ebd-c2e4-4c15-8657-4d40cc7626f1',
      source: 'DR',
      companyDetails: {
        companyId: '74473',
        companyName: 'Monzo',
        url: 'https://monzo.com',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
    {
      id: '89c20324-1390-4ea2-acbc-7cc05a98db59',
      source: 'CRB',
      companyDetails: {
        companyId: '33164743-a09a-45cb-a160-a54ace4dc3da',
        companyName: 'Monzon Biomed',
        url: 'http://www.monzonbiomed.com/',
        countryName: 'Spain',
        countryCode: 'ES',
      },
    },
    {
      id: '807a6c17-a4ff-4b74-b9ef-fee94364f451',
      source: 'CRB',
      companyDetails: {
        companyId: '456754eb-67c6-98db-780a-327db4c05c8f',
        companyName: 'monzoom.com',
        url: 'https://www.monzoom.com',
        countryName: 'Austria',
        countryCode: 'AT',
      },
    },
    {
      id: '0511b523-c25b-443b-932f-4d501d5c0cd4',
      source: 'CRB',
      companyDetails: {
        companyId: '09c52685-0f60-4a6f-a5fc-5c51ae5bec3a',
        companyName: 'Monzoon Networks',
        url: 'http://hotspot.monzoon.net',
        countryName: 'Switzerland',
        countryCode: 'CH',
      },
    },
  ],
  internal: [
    {
      //replace property id by company_id
      //TODO: change all records
      company_id: '6',
      source: 'MANUAL',
      companyDetails: {
        companyId: '6',
        companyName: 'MonzoBCG',
        url: 'www.MonzoBCG.com',
        countryName: 'England',
        countryCode: null,
      },
    },
    {
      id: '127237',
      source: 'DR',
      companyDetails: {
        companyId: '127237',
        companyName: 'Monzo',
        url: 'https://monzo.com',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
    {
      id: '141132',
      source: 'MANUAL',
      companyDetails: {
        companyId: '141132',
        companyName: 'Monzo',
        url: 'https://monzo.com/',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
  ],
}
