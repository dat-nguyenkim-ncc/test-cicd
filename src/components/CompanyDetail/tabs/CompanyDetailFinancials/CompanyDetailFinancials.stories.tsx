import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import CompanyDetailFinancials, { CompanyDetailFinancialsProps } from './CompanyDetailFinancials'
import { companyFinancials } from '../../../../__mock__'
import { Section } from '../../../primitives'

export default {
  title: 'Company Details/CompanyDetailFinancials',
  component: CompanyDetailFinancials,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta

const Template: Story<CompanyDetailFinancialsProps> = args => (
  <Section sx={{ m: 6, mx: 'auto' }}>
    <CompanyDetailFinancials {...args} />
  </Section>
)

export const Default = Template.bind({})
Default.args = {
  data: companyFinancials(),
} as CompanyDetailFinancialsProps
