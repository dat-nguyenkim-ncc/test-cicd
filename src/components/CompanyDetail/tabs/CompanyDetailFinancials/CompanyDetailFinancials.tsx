import React, { useState } from 'react'
import { Box, Grid, Link } from 'theme-ui'
import strings from '../../../../strings'
import { formatMoney } from '../../../../utils'
import CompanyDetailInline from '../CompanyDetailInline'
import { Paragraph } from '../../../primitives'
import { CompanyFinancials, InvestorsProps, ValueCurrency } from '../../../../types'
import moment from 'moment'
import CompanyDetailInlineInvestors from '../CompanyDetailInlineInvestors'
import { SortByDate } from '../../../../pages/CompanyForm/helpers'
import TotalFunding from './TotalFunding'
import { CompanyLink } from '../../..'
import { DEFAULT_VIEW_DATE_FORMAT } from '../../../../utils/consts'

export type CompanyDetailFinancialsProps = {
  data: CompanyFinancials
  isInvestment?: boolean
}

type ShowTextProps = {
  value: string
  max: number
}
const ShowText = ({ value, max }: ShowTextProps) => {
  const [showFull, setShowFull] = useState(false)
  return (
    <>
      <Paragraph sx={{ mb: 2, wordWrap: 'break-word' }}>
        {showFull ? value : `${value.slice(0, max)}${value.length > max ? '...' : ''}`}
      </Paragraph>
      {value.length >= max && (
        <Link
          href=""
          onClick={e => {
            e.preventDefault()
            setShowFull(!showFull)
          }}
        >
          {showFull ? 'Show less' : 'Show more'}
        </Link>
      )}
    </>
  )
}

const CompanyDetailFinancials = ({ data, isInvestment = false }: CompanyDetailFinancialsProps) => {
  const {
    companyDetail: { financials: copy },
  } = strings

  const mapTitles = {
    valuation: copy.valuation,
    fundingTotal: copy.fundingTotal,
    fundingStage: copy.fundingStage,
    lastFundingAt: copy.lastFundingAt,
    lastFundingDate: copy.lastFundingDate,
    equityFundingTotal: copy.equityFundingTotal,
    leadInvestor: copy.leadInvestor,
    lastFundingAmount: copy.lastFundingAmount,
  }

  const tableKeys = React.useMemo(
    () => [
      {
        key: 'date',
        format: (v: string | number) =>
          moment(v).format(DEFAULT_VIEW_DATE_FORMAT) !== 'Invalid date' ? (
            <Paragraph>{moment(v).format(DEFAULT_VIEW_DATE_FORMAT)}</Paragraph>
          ) : (
            <Paragraph>Not available</Paragraph>
          ),
      },
      {
        key: 'roundTypes',
        format: (rounds: string[]) =>
          rounds.map((value: string, index) => (
            <Paragraph key={index}>{`${value}${index < rounds.length - 1 ? ',' : ''}`}</Paragraph>
          )),
      },
      {
        key: data.isExternalViewDetail ? 'investment' : 'investment_usd',
        format: (v: ValueCurrency) =>
          v.value ? (
            <Paragraph>{formatMoney(v.value, v.currency)}</Paragraph>
          ) : (
            <Paragraph>Not available</Paragraph>
          ),
      },
      {
        key: data.isExternalViewDetail ? 'valuation' : 'valuation_usd',
        format: (v: ValueCurrency) =>
          v.value ? (
            <Paragraph>{formatMoney(v.value, v.currency)}</Paragraph>
          ) : (
            <Paragraph>Not available</Paragraph>
          ),
      },
      isInvestment
        ? {
            key: 'company',
            format: (company: { company_id: number; name: string; logo_bucket_url?: string }) =>
              !!company?.company_id ? (
                <CompanyLink company={company} />
              ) : (
                <Paragraph>Not available</Paragraph>
              ),
          }
        : {
            key: 'investors',
            format: ({ lead, other }: InvestorsProps) => (
              <CompanyDetailInlineInvestors lead={lead} other={other} />
            ),
          },
      {
        key: 'comment',
        format: (value: string) => <ShowText value={value} max={120} />,
      },
    ],
    [data.isExternalViewDetail, isInvestment]
  )

  type FundingRoudsKeys = keyof typeof copy.fundingRounds

  const showFundingRound = (data: any) => {
    return (
      moment(data.date).format(DEFAULT_VIEW_DATE_FORMAT) !== 'Invalid date' ||
      data.investment.value ||
      data.roundTypes.length ||
      data.valuation.value ||
      data.investors.lead ||
      data.investors.other
    )
  }

  return (
    <Box>
      <Grid mt={4} gap={0} columns={'50% 50%'}>
        {Object.keys(mapTitles).map(t => {
          if (!mapTitles[t as keyof typeof mapTitles]) return null

          const title = `${mapTitles[t as keyof typeof mapTitles]}:`
          let detail = data[t as keyof typeof data]
          if (!detail) return null

          detail =
            (detail as ValueCurrency).value && (detail as ValueCurrency).currency
              ? formatMoney((detail as ValueCurrency).value, (detail as ValueCurrency).currency)
              : detail.toString()

          if (detail === '[object Object]') return null

          return <CompanyDetailInline key={title} title={title} detail={detail} />
        })}
      </Grid>
      {data.fundingRounds && data.fundingRounds.length > 0 && (
        <Box mt={!isInvestment ? 5 : 0}>
          {!isInvestment && (
            <TotalFunding
              isExternalViewDetail={data.isExternalViewDetail}
              fundingRounds={data.fundingRounds}
            />
          )}
          <Paragraph bold>{copy.fundingRounds.heading}</Paragraph>
          <Grid sx={{ p: 5 }} mt={4} columns={'1fr 1fr 1fr 1fr 1fr 1fr'}>
            {tableKeys.map((t, index) => (
              <Paragraph sx={{ gridColumn: index + 1 }} key={t.key} bold>
                {copy.fundingRounds[t.key as FundingRoudsKeys]}
              </Paragraph>
            ))}
          </Grid>
          {data.fundingRounds
            .slice()
            .sort((a, b) => SortByDate(a, b, 'date', 'descending'))
            .filter(f => !!showFundingRound(f))
            .map((item, line) => {
              const f = {
                ...item,
                valuation_usd: item.valuation,
                investment_usd: item.investment,
              }
              return (
                <Grid
                  sx={{
                    p: 5,
                    backgroundColor: line % 2 === 0 ? 'gray03' : 'transparent',
                    borderRadius: 10,
                  }}
                  key={line}
                  mt={4}
                  columns={'1fr 1fr 1fr 1fr 1fr 1fr'}
                >
                  {tableKeys.map((t, index) => {
                    // @ts-ignore
                    const value = f[t.key] || ''

                    return t.format ? (
                      <Box sx={{ gridColumn: index + 1 }} key={index}>
                        {t.format(value)}
                      </Box>
                    ) : (
                      <Paragraph sx={{ gridColumn: index + 1 }} key={index}>
                        {value || '-'}
                      </Paragraph>
                    )
                  })}
                </Grid>
              )
            })}
        </Box>
      )}
    </Box>
  )
}

export default CompanyDetailFinancials
