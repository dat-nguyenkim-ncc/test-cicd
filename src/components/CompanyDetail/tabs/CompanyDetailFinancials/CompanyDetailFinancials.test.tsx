import * as React from 'react'
import { shallow } from 'enzyme'
import CompanyDetailFinancials from './CompanyDetailFinancials'
import { companyFinancials } from '../../../../__mock__'

describe('CompanyDetailFinancials', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<CompanyDetailFinancials data={companyFinancials()} />)
    expect(wrapper).toMatchSnapshot()
  })
})
