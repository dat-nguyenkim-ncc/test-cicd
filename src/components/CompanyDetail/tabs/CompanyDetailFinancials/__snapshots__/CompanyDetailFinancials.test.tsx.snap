// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyDetailFinancials matches snapshot 1`] = `ShallowWrapper {}`;
